packages/server/parseFormBody.ts

```diff
@@ -0,0 +1,52 @@
+import {APP_MAX_AVATAR_FILE_SIZE} from 'parabol-client/utils/constants'
+import {HttpResponse, getParts, MultipartField} from 'uWebSockets.js'
+
+interface Uploadable {
+  contentType: string
+  buffer: Buffer
+}
+
+type ParseFormBodySignature = {
+  res: HttpResponse
+  contentType: string
+}
+
+const parseRes = (res: HttpResponse) => {
+  return new Promise<Buffer | null>((resolve) => {
+    let buffer: Buffer
+    res.onData((ab, isLast) => {
+      const curBuf = Buffer.from(ab)
+      buffer = buffer ? Buffer.concat([buffer, curBuf]) : isLast ? curBuf : Buffer.concat([curBuf])
+      if (buffer.length > APP_MAX_AVATAR_FILE_SIZE) resolve(null)
+      if (isLast) resolve(buffer)
+    })
+  })
+}
+
+const parseFormBody = async ({res, contentType}: ParseFormBodySignature): Promise<JSON | null> => {
+  const resBuffer = await parseRes(res)
+  if (!resBuffer) return null
+  const parts = getParts(resBuffer, contentType) as MultipartField[]
+  if (!parts) return null
+
+  try {
+    let parsedBody
+    parts.forEach(({name, data, type}) => {
+      if (name === 'body') {
+        parsedBody = JSON.parse(Buffer.from(data).toString())
```

Comment:
parsing the string could throw an error

Reply:
this is true. the entire `forEach` is currently contained within a try-catch block. do you think a nested try-catch block inside the loop should be added?

my thinking is that we expect a parsable body field, so if the client passes an un-parseable body field, we should immediately throw an error which will ultimate return 422 client error.

Reply:
good thinking, yeah, a single try/catch should work, if the client fails that they're probably up to no good anyways.